napari/_vispy/canvas.py

```diff
@@ -112,12 +119,24 @@ def __init__(
         self._scene_canvas = NapariSceneCanvas(
             *args, keys=None, vsync=True, **kwargs
         )
+
         self.view = self.central_widget.add_view(border_width=0)
         self.camera = VispyCamera(
             self.view, self.viewer.camera, self.viewer.dims
         )
+
+        self.grid = self.central_widget.add_grid(
+            border_width=0, spacing=viewer.grid.spacing
+        )
+        self.grid._default_class = NapariViewBox
+        self.grid_views = []
+        self.grid_cameras = []
+
         self.layer_to_visual: dict[Layer, VispyBaseLayer] = {}
-        self._overlay_to_visual: dict[Overlay, VispyBaseOverlay] = {}
+        self._overlay_to_visuals: dict[Overlay, list[VispyBaseOverlay]] = {}
```

Comment:
`list[VispyBaseOverlay]` this is list of one per viewbox?

Reply:
basically, yes, if the grid mode is on, otherwise it's a list of 1 element.

Reply:
Then, I think, you should add a comment with an explanation for future readers.